sdk/core/System.ClientModel/README.md

```diff
@@ -23,71 +22,290 @@ dotnet add package System.ClientModel
 
-A very basic client implementation might use the following approach:
+**Protocol method** are low-level methods that take parameters that correspond to the service HTTP API and return a `ClientResult` holding only the raw HTTP response details.  These methods also take an optional `RequestOptions` parameter that allows the client pipeline and the request to be configured for the duration of the call.
 
-```csharp
-ApiKeyCredential credential = new ApiKeyCredential(key);
-ApiKeyAuthenticationPolicy authenticationPolicy = ApiKeyAuthenticationPolicy.CreateBearerAuthorizationPolicy(credential);
-ClientPipeline pipeline = ClientPipeline.Create(pipelineOptions, authenticationPolicy);
+The following sample illustrates how to call a convenience method and access both the strongly-typed output model and the details of the HTTP response.
 
-PipelineMessage message = pipeline.CreateMessage();
-message.Apply(requestOptions);
-message.MessageClassifier = PipelineMessageClassifier.Create(stackalloc ushort[] { 200 });
+```C# Snippet:ClientResultTReadme
+// Create a client
+string? key = Environment.GetEnvironmentVariable("MAPS_API_KEY");
+ApiKeyCredential credential = new(key!);
+MapsClient client = new(new Uri("https://atlas.microsoft.com"), credential);
 
-PipelineRequest request = message.Request;
-request.Method = "GET";
-request.Uri = new Uri("https://www.example.com/");
-request.Headers.Add("Accept", "application/json");
+// Call a service method, which returns ClientResult<T>
+IPAddress ipAddress = IPAddress.Parse("2001:4898:80e8:b::189");
+ClientResult<IPAddressCountryPair> result = await client.GetCountryCodeAsync(ipAddress);
 
-pipeline.Send(message);
-Console.WriteLine(message.Response.Status);
+// ClientResult<T> has two members:
+//
+// (1) A Value property to access the strongly-typed output
+IPAddressCountryPair value = result.Value;
+Console.WriteLine($"Country is {value.CountryRegion.IsoCode}.");
+
+// (2) A GetRawResponse method for accessing the details of the HTTP response
+PipelineResponse response = result.GetRawResponse();
+
+Console.WriteLine($"Response status code: '{response.Status}'.");
+Console.WriteLine("Response headers:");
+foreach (KeyValuePair<string, string> header in response.Headers)
+{
+    Console.WriteLine($"Name: '{header.Key}', Value: '{header.Value}'.");
+}
+```
+
+For more information on client service methods, see [Client service method samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/samples/ServiceMethods.md).
+
+### Handling exceptions that result from failed requests
+
+When a service call fails, service clients throw a `ClientResultException`.  The exception exposes the HTTP status code and the details of the service response if available.
```

Comment:
It would be good to clarify what is considered a failure, i.e. status code deemed error by the classifier. In general, maybe it's a good section to discuss the classifier and how it sets the IsError property on the response.

Reply:
ResponseClassifier feels like a client-author topic, and this is more of a client-user section.  But, I can add the discussion of the classifier in the client-author section and link back to it here.

We're not currently exposing the message classifier on RequestOptions, but that shouldn't be too hard to move into ClientModel when we want that.  It'll require a generator update.

Reply:
Addressed